### PR TITLE
Completed exercise big-o-notation. - Jesus Araujo 

### PR DESCRIPTION
**Pull request by j-alejandro-araujo**
Screenshot 2023-06-20 at 9 02 34 PM

### CODE REVIEW

File: big-o-notation/big-o.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+function uniqueLinear(words) {
+  const seen = {}; // _ * _ = O(?) --> 1 * 1 = O(1)
+  const unique = []; // _ * _ = O(?) --> 1 * 1 = O(1)
+  for (
+    let i = 0; // _ * _ = O(?) --> 1 * 1 = O(1)
+    i < words.length; // _ * _ = O(?) --> 2 * n = O(n)
+    i++ // _ * _ = O(?) --> 2 * n = O(n)
+  ) {
+    const word = words[i]; // _ * _ = O(?) --> 2 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:11]: Should be `O(n)`

File: big-o-notation/big-o.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if (!seen[word]) {
+      // _ * _ = O(?) --> 2 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:13]: Should be `O(n)`

File: big-o-notation/big-o.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      seen[word] = true; // _ * _ = O(?) --> 2 * 1 = O(1)
+      unique[unique.length] = word; // _ * _ = O(?) --> 3 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:15]: Should be `O(n)`

File: big-o-notation/big-o.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+} // Big O Notation for uniqueLinear: O(?) --> O(n)
+
+function uniqueQuadratic(words) {
+  const unique = []; // _ * _ = O(?) --> 1 * 1 = O(1)
+  for (
+    let i = 0; // _ * _ = O(?) --> 1 * 1 = O(1)
+    i < words.length; // _ * _ = O(?) --> 2 * n = O(n)
+    i++ // _ * _ = O(?) --> 2 * n = O(n)
+  ) {
+    const word = words[i]; // _ * _ = O(?) --> 2 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:28]: Should be `O(n)`

File: big-o-notation/big-o.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    let isUnique = true; // _ * _ = O(?) --> 1 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:29]: Should be `O(n)`

File: big-o-notation/big-o.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    for (
+      let c = 0; // _ * _ = O(?) --> 1 * n = O(n)
+      c < i; // _ * _ = O(?) --> 1 * n * n = O(n^2)
+      c++ // _ * _ = O(?) --> 2 * n * n = O(n^2)
+    ) {
+      const comparing = words[c]; // _ * _ = O(?) --> 2 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:35]: Should be `O(n^2)`

File: big-o-notation/big-o.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      if (comparing === word) {
+        // _ * _ = O(?) --> 1 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:37]: Should be `O(n^2)`

File: big-o-notation/big-o.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        isUnique = false; // _ * _ = O(?) --> 1 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:38]: Should be `O(n^2)`

File: big-o-notation/big-o.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      }
+    }
+    if (isUnique) {
+      // _ * _ = O(?) --> 1 * 1 = O(1)
```
Comment (shawnkost) [inline big-o-notation/big-o.js:42]: Should be `O(n)`

**Review by shawnkost — CHANGES_REQUESTED** [review]
Few changes requested here!

**Comment by j-alejandro-araujo** [comment]
Adjustments complete, thank you!

File: big-o-notation/big-o.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  const unique = []; // _ * _ = O(?) --> 1 * 1 = O(1)
+  for (
+    let i = 0; // _ * _ = O(?) --> 1 * 1 = O(1)
+    i < words.length; // _ * _ = O(?) --> 2 * n = O(n)
+    i++ // _ * _ = O(?) --> 2 * n = O(n)
+  ) {
+    const word = words[i]; // _ * _ = O(?) --> 2 * n = O(n)
+    if (!seen[word]) {
+      // _ * _ = O(?) --> 2 * n = O(n)
+      seen[word] = true; // _ * _ = O(?) --> 2 * 1 = O(1)
```
Comment (TimelikeClosure) [inline big-o-notation/big-o.js:14]: Should be `O(n)`.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: big-o-notation/big-o.js (hunk 10 above)
Comment (j-alejandro-araujo) [reply big-o-notation/big-o.js]: Fixed!

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**